lib/fathead/pandas/parse.py

```diff
@@ -0,0 +1,204 @@
+        content = []
+        for paragraph in paragraphs:
+            content.append(paragraph.get_text())
+        return '\n\n'.join(content)
+
+    def parse_for_first_paragraph(self, section):
+        """
+        Returns the first paragraph of text for a given function
+        Fixes up double spacing and newlines.
+        Args:
+            section: A section of parsed HTML that represents a function definition
+        Returns:
+            First paragraph found with text
+        """
+        paragraphs = section.find_all('p')
+        for paragraph in paragraphs:
+            if paragraph.text:
+                return paragraph.text.replace('  ', ' ').replace('\n', ' ').replace('\\n', r'\\n')
+        return ''
+
+    def get_data(self):
+        """
+        Get the parsed data.
+        Returns:
+            self.parsed_data: Dict containing necessary data elements
+        """
+        return self.parsed_data
+
+
+class PythonDataOutput(object):
+    """
+    Object responsible for outputting data into the output.txt file
+    """
+    def __init__(self, data, output_file):
+        self.data = data
+        self.output_file = output_file
+
+    def truncate(self, text, length):
+        if len(text) <= length:
+            return text
+        return "{}...".format(text[:length].rsplit(' ', 1)[0])
+
+    def create_file(self):
+        """
+        Iterate through the data and create the needed output.txt file, appending to file as necessary.
+        """
+        for data_element in self.data:
+            title = data_element.get('title')
+            abstract = self.truncate(data_element.get('first_paragraph'), 400)
+            example = data_element.get('example')
+            if example:
+                abstract = "{}<br>{}".format(abstract, example)
+            abstract = abstract.replace('\n', '\\n')
+            anchor = data_element.get('anchor')
+
+            list_of_data = [
+                title,                      # unique name
+                'A',                        # type is article
+                '',                         # no redirect data
+                '',                         # ignore
+                '',                         # no categories
+                '',                         # ignore
+                '',                         # no related topics
+                '',                         # ignore
+                URL_ROOT,                   # add an external link back to Scikit home
+                '',                         # no disambiguation
+                '',                         # images
+                abstract,                   # abstract
+                anchor                      # url to doc
+            ]
+
+            self.output_file.write('{}\n'.format('\t'.join(list_of_data)))
+
+
+if __name__ == "__main__":
+    with open('output.txt', 'w') as output_file:
+        for dir_path, dir_name, file_names in os.walk(DOWNLOADED_HTML_PATH):
```

Comment:
@souravsingh `DOWNLOADED_HTML_PATH` is not defined. I would appreciate if you try running the code on your system once and make final changes before next review.